site/content/en/development/design-docs/pls-integration.md

```diff
@@ -0,0 +1,91 @@
+| `service.beta.kubernetes.io/azure-pls-ip-configuration-ip-address-count` | `[1-8]`                            | Total number of private NAT IPs to allocate. | Optional | 1 |
+| `service.beta.kubernetes.io/azure-pls-ip-configuration-ip-address`       | `"10.0.0.7 ... 10.0.0.10"`         | A space separated list of static IPs to be allocated. Total number of IPs should not be greater than the ip count specifed in `service.beta.kubernetes.io/azure-pls-ip-configuration-ip-address-count`. If there are fewer IPs specified, the rest are dynamically allocated. The first IP in the list is set as `Primary`. |  Optional | All IPs are dynamically allocated. |
+| `service.beta.kubernetes.io/azure-pls-fqdns`                             | `"fqdn1 fqdn2"`                    | A space separated list of fqdns associated with the PLS. | Optional | `[]` |
+| `service.beta.kubernetes.io/azure-pls-proxy-protocol`                    | `"true"` or `"false"`              | Boolean indicatin whether the TCP Proxy protocol needs to be enabled on the PLS to determine true source IP address. | Optional | `false` |
+| `service.beta.kubernetes.io/azure-pls-visibility`                        | `"sub1 sub2 sub3 … subN"` or `"*"` | A space separated list of Azure subscription ids for which the private link service is visible. Use `"*"` to expose the PLS to all subs (Least restrictive). | Optional | Empty list `[]` indicating role-based access control only: This private link service will only be available to individuals with role-based access control permissions within your directory. (Most restrictive) |
+| `service.beta.kubernetes.io/azure-pls-auto-approval`                     | `"sub1 sub2 sub3 … subN"`          | A space separated list of Azure subscription ids. This allows PE connection requests from the subscriptions listed to the PLS to be automatically approved. |  Optional | `[]` |
+
+For more details about each configuration, please refer to [Azure Private Link Service Documentation](https://docs.microsoft.com/en-us/cli/azure/network/private-link-service?view=azure-cli-latest#az-network-private-link-service-create).
+
+
+## Design Details
+
+### Creating AKS managed PrivateLinkService
+
+When a `LoadBalancer` typed service is created, a LB frontend IP configuration is created with either a dynamically generated or user specified IP. Or if the service has `loadBalancerIP` in its spec, an existing LB frontend IP configuration may be reused. When service has `service.beta.kubernetes.io/azure-pls-create` set to `true`, a PLS resource attached to the LB frontend is created in the `MC_` resource group for the AKS cluster. 
+
+The Kubernetes service creating the PLS is assigned as the owner of the resource. Azure cloud provider tags the PLS with service name `<namespace>\<service name>`. Only the owner service can later update the properties of the PLS resource.
```

Comment:
for the tag mentioned here, could you also add its key here, e.g. `"k8s-azure-service"`?

Reply:
I take a look at how we name the PIP labels: `kubernetes-dns-label-service: default\nginx`, so I'm now following similar pattern, `kubernetes-owner-service: default\nginx`. Updated doc.